lib/rules/simplify-set-operations.ts

```diff
@@ -0,0 +1,419 @@
+                            (e) => e !== negateOperand,
+                        )
+                        let fixedLeftText = others.map((e) => e.raw).join("&&")
+                        if (others.length >= 2) {
+                            // Wrap with brackets
+                            fixedLeftText = `[${fixedLeftText}]`
+                        }
+                        const fixedRightText = getRawTextToNot(negateOperand)
+                        return `[${
+                            eccNode.negate ? "^" : ""
+                        }${`${fixedLeftText}--${fixedRightText}`}]`
+                    },
+                })
+            }
+
+            /**
+             * Checks the given expression and reports whether it can be converted to intersection by reducing its complement.
+             * Returns true if reported.
+             *
+             * e.g.
+             * - `[a--[^b]]` -> `[a&&b]`
+             */
+            function toIntersection(
+                expression: ClassIntersection | ClassSubtraction,
+                expressionRight:
+                    | ClassIntersection
+                    | ClassSubtraction
+                    | ClassSetOperand
+                    | null,
+                eccNode: ExpressionCharacterClass,
+            ) {
+                if (expression.type !== "ClassSubtraction") {
+                    return false
+                }
+                const { left, right } = expression
+                if (!isNegate(right)) {
+                    return false
+                }
+                return reportWhenFixedIsCompatible({
+                    reportNode: expression,
+                    targetNode: eccNode,
+                    messageId: "toIntersection",
+                    fix() {
+                        let fixedLeftText = left.raw
+                        if (left.type === "ClassSubtraction") {
+                            // Wrap with brackets
+                            fixedLeftText = `[${fixedLeftText}]`
+                        }
+                        const fixedRightText = getRawTextToNot(right)
+                        let fixedText = `${fixedLeftText}&&${fixedRightText}`
+
+                        if (expressionRight) {
+                            // Wrap with brackets
+                            fixedText = `[${fixedText}]`
+                        }
+                        const targetRaw = eccNode.raw
+                        return `${targetRaw.slice(
+                            0,
+                            expression.start - eccNode.start,
+                        )}${fixedText}${targetRaw.slice(
+                            expression.end - eccNode.start,
+                        )}`
+                    },
+                })
+            }
+        }
+
+        return defineRegexpVisitor(context, {
+            createVisitor,
+        })
+    },
+})
+
+/** Gets the parsed result element. */
+function getParsedElement(
```

Comment:
Could we move this function up before `export default createRule(...)`? We have `export default createRule(...)` as the last statement in all other rules (I think), so I would like to keep it consistent.

Reply:
Yeah. I need to move it up.